### ISSUE TITLE
<Image /> with borderWidth rendering weird content on Android

### ISSUE THREAD

**Issue by kientux** [facebook/react-native#19667]
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [x] Use the latest React Native version: https://github.com/facebook/react-native/releases
  - [x] Run `react-native info` in your terminal and paste its contents under "Environment"
  - [x] Let us know how to reproduce the issue. Include a code sample, share a project, or
    share an app that reproduces the issue using https://snack.expo.io/

## Environment
```
Environment:
  OS: macOS High Sierra 10.13.5
  Node: 8.11.2
  Yarn: 1.7.0
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.4 Build version 9F1027a
  Android Studio: 3.1 AI-173.4697961

Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: ~0.55.2 => 0.55.4
```

## Description
On Android device/simulator, `Image` component with `borderWidth` style renders weird content, with unexpected behaviours. 

## Steps to Reproduce
This is a very simple app will reproduce the bug (fresh project created by `create-react-native-app`):

```javascript
import React from "react";
import { StyleSheet, View, Image } from "react-native";

export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Image
          source={require("./product.png")}
          style={{
            width: 128,
            height: 128,
            borderWidth: 1,
            borderColor: "red"
          }}
          resizeMode="center"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  }
});
```

## Expected Behavior
The image must be displayed correctly like on iOS.

## Actual Behavior
For example, it causes a vertical line:

![screen shot 2018-06-12 at 15 59 54](https://user-images.githubusercontent.com/4200743/41280689-c577c938-6e59-11e8-800d-22fcf2324e35.png)

This is the original `product.png`: ![product](https://user-images.githubusercontent.com/4200743/41280219-9d4553a0-6e58-11e8-8c26-6ac6f548928e.png)

`resizeMode="center"` is just for easier to see the black line. The line is in there with every `resizeMode`, but if I set it to `cover` or `contain`, etc., image will overlap the line so we won't see it.

With a different image, Android on the left, iOS on the right, and actual image for reference:

![screen shot 2018-06-12 at 16 20 03](https://user-images.githubusercontent.com/4200743/41281783-80517c2a-6e5c-11e8-9037-022fb64179bf.png)

Or this:

![screen shot 2018-06-12 at 16 31 39](https://user-images.githubusercontent.com/4200743/41282387-241003d0-6e5e-11e8-857a-504312fb3b29.png)

I've searched in issues to find if someone had the same problem, and expect to easily found something because this seems very obvious, but ended up found nothing. Is there something wrong with my project setup? This happens without Expo too.

**Comment by kientux** [comment]
I want to add some more info:
- This happens only for image from `require('./path/to/local/image.png')`, not from `{ uri: 'https://website.example/image.png' }`
- This will causes `borderRadius` of `Image` component stop working (on Android, of course)

**Comment by kientux** [comment]
Okay so this is fixed in React Native v0.56.0. I will close this.
